SpatialGDKEditorToolbar/Source/Private/SpatialGDKEditorGenerateSnapshot.cpp

```diff
@@ -482,7 +468,37 @@ bool SpatialGDKGenerateSnapshot(UWorld* World)
 		return false;
 	}
 
+	return true;
+}
+
+bool SpatialGDKGenerateSnapshot(UWorld* World)
+{
+	const USpatialGDKEditorToolbarSettings* Settings = GetDefault<USpatialGDKEditorToolbarSettings>();
+	FString SavePath = FPaths::Combine(Settings->GetSpatialOSSnapshotPath(), Settings->GetSpatialOSSnapshotFile());
+	if (!ValidateAndCreateSnapshotGenerationPath(SavePath))
+	{
+		return false;
+	}
+
+	UE_LOG(LogSpatialGDKSnapshot, Display, TEXT("Saving snapshot to: %s"), *SavePath);
+
+	Worker_ComponentVtable DefaultVtable{};
+	Worker_SnapshotParameters Parameters{};
+	Parameters.default_component_vtable = &DefaultVtable;
+
+	bool bSuccess = true;
+	Worker_SnapshotOutputStream* OutputStream = Worker_SnapshotOutputStream_Create(TCHAR_TO_UTF8(*SavePath), &Parameters);
```

Comment:
I assume that the output stream creation can't fail? if not maybe we should check for it being nullptr?